OrbitVulkanLayer/QueueManager.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "QueueManager.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_vulkan_layer {
+
+void QueueManager::TrackQueue(const VkQueue& queue, const VkDevice& device) {
+  absl::WriterMutexLock lock(&mutex_);
+  CHECK(!queue_to_device_.contains(queue) || queue_to_device_.at(queue) == device);
+  queue_to_device_[queue] = device;
```

Comment:
I personally would use `queue_to_device_.emplace(queue, device)` to make it clear that you are not actually overwriting (in case `queue_to_device_.at(queue) == device`) (but this is super minor).

Reply:
Done, thanks!